code/workhorse/screen_index_helper.R

```diff
@@ -0,0 +1,142 @@
+compute_covariance_h <- function(simulator,set1,set2){
+  #set2 can be pool or benchmark or the merge of these two
+  # set1= as.matrix(set1)
+  # set2 = as.matrix(set2)
+  
+  d = simulator$model[["numDim"]]
+  beta = simulator$model[["beta"]]
+  sigma = simulator$model[["sig2"]]
+  number = nrow(set1)
+  newcov=matrix(nrow=dim(set2)[1],ncol=number)
+  
+  for ( i in 1:dim(set2)[1]) {
+    for (j in 1:number) {
+      dist= -sum((set2[i,1:d]-set1[j,1:d])^2*beta)
+      newcov[i,j]= sigma*exp(dist)
+    }
+  }
+  cat("compute kernel done \n")
+  return(newcov)
+}
+
+# you have three metrics, so you should have three functions to compute the metrics.
+# This is the outline
+predict_testset <- function(simulator,testset, unfold=FALSE) {
+  #unfold is true, return the mean and covariance.
+  sigma = simulator$model[["sig2"]]
+  inverse=simulator$model$invVarMatrix
+  mu=simulator$model[["mu"]]
+  fitZ = simulator$model$Z
+  nugget= simulator$model[["nugget"]]
```

Comment:
These and the relevant isage can probably go in a function. You may not need these variabls.

Reply:
This is related to the principle we discussed in our code session whether we used a temporary variable or not. If I don't, the name will be very long whenever I need to use that one field of the data of the object.